failure during compilation of kernel binary with SBCL

I have tried integrating maxima-jupyter with the Gentoo package management system. In the past I was able to build it successfully but today I have discovered a problem that I cannot solve because I don't know anything about LISP. 

In the installation script I download and install locally the latest version of quicklisp (to /var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install/) and use the following building script:

```
:lisp (load "/var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install/setup.lisp")
:lisp (load "/var/tmp/portage/dev-python/maxima-jupyter-9999/work/maxima-jupyter-9999/load-maxima-jupyter.lisp")
:lisp (sb-ext:save-lisp-and-die #P"/var/tmp/portage/dev-python/maxima-jupyter-9999/work/maxima-jupyter-9999/maxima-jupyter-exec" :executable t :toplevel 'maxima-jupyter:kernel-start-exec
```

and finally I recall it with

`
MAXIMA_PREFIX=/usr maxima -l $CL_BINARY -b install-script.lisp
`

Here is the full trace log:

```
Maxima 5.42.1 http://maxima.sourceforge.net
using Lisp SBCL 1.4.9
Distributed under the GNU Public License. See the file COPYING.
Dedicated to the memory of William Schelter.
The function bug_report() provides bug reporting information.
(%i1) batch("/var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install-script.lisp")

read and interpret file: /var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install-script.lisp
TTo load "maxima-jupyter":
  Load 1 ASDF system:
    maxima-jupyter
; Loading "maxima-jupyter"
To load "alexandria":
  Load 1 ASDF system:
    asdf
  Install 1 Quicklisp release:
    alexandria
; Fetching #<URL "http://beta.quicklisp.org/archive/alexandria/2018-12-10/alexandria-20181210-git.tgz">
; 50.00KB
==================================================
51,197 bytes in 0.03 seconds (1785.61KB/sec)
; Loading "alexandria"
[package alexandria.0.dev].....................
; Loading "maxima-jupyter"
To load "common-lisp-jupyter":
  Load 3 ASDF systems:
    alexandria asdf uiop
  Install 19 Quicklisp releases:
    asdf-system-connections babel bordeaux-threads cffi
    cl-base64 cl-containers closer-mop common-lisp-jupyter
    ironclad iterate jsown metatilities-base nibbles pzmq
    trivial-features trivial-gray-streams trivial-mimes
    trivial-utf-8 uuid
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-utf-8/2011-10-01/trivial-utf-8-20111001-darcs.tgz">
; 5.91KB
==================================================
6,055 bytes in 0.00 seconds (0.00KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/uuid/2018-08-31/uuid-20180831-git.tgz">
; 21.71KB
==================================================
22,230 bytes in 0.01 seconds (3101.28KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-mimes/2018-08-31/trivial-mimes-20180831-git.tgz">
; 22.07KB
==================================================
22,597 bytes in 0.01 seconds (2206.74KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-gray-streams/2018-10-18/trivial-gray-streams-20181018-git.tgz">
; 7.84KB
==================================================
8,024 bytes in 0.00 seconds (0.00KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cffi/2018-12-10/cffi_0.20.0.tgz">
; 253.99KB
==================================================
260,085 bytes in 0.15 seconds (1649.28KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/pzmq/2019-05-21/pzmq-20190521-git.tgz">
; 19.61KB
==================================================
20,076 bytes in 0.01 seconds (2178.39KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/jsown/2017-11-30/jsown-20171130-git.tgz">
; 14.02KB
==================================================
14,360 bytes in 0.00 seconds (2804.69KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/iterate/2018-02-28/iterate-20180228-git.tgz">
; 325.97KB
==================================================
333,795 bytes in 0.14 seconds (2328.37KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/nibbles/2018-08-31/nibbles-20180831-git.tgz">
; 17.23KB
==================================================
17,647 bytes in 0.01 seconds (2461.91KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/ironclad/2019-03-07/ironclad-v0.45.tgz">
; 1415.85KB
==================================================
1,449,826 bytes in 0.63 seconds (2250.95KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/closer-mop/2019-05-21/closer-mop-20190521-git.tgz">
; 22.16KB
==================================================
22,696 bytes in 0.01 seconds (2014.91KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/metatilities-base/2019-05-21/metatilities-base-20190521-git.tgz">
; 68.68KB
==================================================
70,325 bytes in 0.05 seconds (1346.60KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/asdf-system-connections/2017-01-24/asdf-system-connections-20170124-git.tgz">
; 4.82KB
==================================================
4,933 bytes in 0.00 seconds (0.00KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-containers/2017-04-03/cl-containers-20170403-git.tgz">
; 223.91KB
==================================================
229,279 bytes in 0.11 seconds (2112.31KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/cl-base64/2015-09-23/cl-base64-20150923-git.tgz">
; 8.39KB
==================================================
8,589 bytes in 0.00 seconds (0.00KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/bordeaux-threads/2018-07-11/bordeaux-threads-v0.8.6.tgz">
; 21.43KB
==================================================
21,941 bytes in 0.01 seconds (1947.89KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/trivial-features/2016-12-04/trivial-features-20161204-git.tgz">
; 10.40KB
==================================================
10,654 bytes in 0.00 seconds (5202.15KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/babel/2017-12-27/babel-20171227-git.tgz">
; 247.98KB
==================================================
253,935 bytes in 0.11 seconds (2214.14KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/common-lisp-jupyter/2019-05-21/common-lisp-jupyter-20190521-git.tgz">
; 76.23KB
==================================================
78,055 bytes in 0.05 seconds (1524.51KB/sec)
; Loading "common-lisp-jupyter"
..................................................
[package babel-encodings].........................
[package babel]...................................
..................................................
[package cffi-sys]................................
[package cffi]....................................
..................................................
[package cffi-features]...........................
[package bordeaux-threads]........................
[package cl-base64]...............................
[package metabang.utilities]......................
[package metabang.cl-containers]..................
..................................................
..................................................
[package closer-mop]..............................
[package closer-common-lisp]......................
[package closer-common-lisp-user].................
[package nibbles].................................
[package ironclad]................................
[package ironclad-vm].............................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
[package iterate].................................
[package jsown]...................................
[package pzmq]; x86_64-pc-linux-gnu-gcc -o /var/tmp/portage/sci-mathematics/maxima-5.42.1/homedir/.cache/common-lisp/sbcl-1.4.9-linux-x64/var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install/dists/quicklisp/software/pzmq-20190521-git/grovel__grovel-tmpGHU3ALSV.o -c -march=core2 -O2 -pipe -ggdb -g -Wall -Wundef -Wsign-compare -Wpointer-arith -O3 -march=core2 -O2 -pipe -ggdb -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wunused-parameter -fno-omit-frame-pointer -momit-leaf-frame-pointer -fno-pie -fPIC -I/var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install/dists/quicklisp/software/cffi_0.20.0/ /var/tmp/portage/sci-mathematics/maxima-5.42.1/homedir/.cache/common-lisp/sbcl-1.4.9-linux-x64/var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install/dists/quicklisp/software/pzmq-20190521-git/grovel__grovel.c
/var/tmp/portage/sci-mathematics/maxima-5.42.1/homedir/.cache/common-lisp/sbcl-1.4.9-linux-x64/var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install/dists/quicklisp/software/pzmq-20190521-git/grovel__grovel.c: In function ‘main’:
/var/tmp/portage/sci-mathematics/maxima-5.42.1/homedir/.cache/common-lisp/sbcl-1.4.9-linux-x64/var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install/dists/quicklisp/software/pzmq-20190521-git/grovel__grovel.c:11:7: warning: unused variable ‘autotype_tmp’ [-Wunused-variable]
   int autotype_tmp;
       ^~~~~~~~~~~~
; x86_64-pc-linux-gnu-gcc -o /var/tmp/portage/sci-mathematics/maxima-5.42.1/homedir/.cache/common-lisp/sbcl-1.4.9-linux-x64/var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install/dists/quicklisp/software/pzmq-20190521-git/grovel__grovel-tmpAAURSO1 -Wl,-O1 -Wl,--as-needed -Wl,--export-dynamic -no-pie /var/tmp/portage/sci-mathematics/maxima-5.42.1/homedir/.cache/common-lisp/sbcl-1.4.9-linux-x64/var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install/dists/quicklisp/software/pzmq-20190521-git/grovel__grovel.o
; /var/tmp/portage/sci-mathematics/maxima-5.42.1/homedir/.cache/common-lisp/sbcl-1.4.9-linux-x64/var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install/dists/quicklisp/software/pzmq-20190521-git/grovel__grovel /var/tmp/portage/sci-mathematics/maxima-5.42.1/homedir/.cache/common-lisp/sbcl-1.4.9-linux-x64/var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install/dists/quicklisp/software/pzmq-20190521-git/grovel__grovel.grovel-tmp.lisp
....................................
[package impl-specific-gray]......................
[package trivial-gray-streams]....................
[package trivial-mimes]...........................
[package trivial-utf-8]...........................
[package uuid]....................................
[package jupyter].................................
[package jupyter-widgets].........................
[package common-lisp-jupyter].....................
[package jupyter-convert].........................
......
; Loading "maxima-jupyter"
[package maxima-jupyter]..;
; caught ERROR:
;   READ error during COMPILE-FILE:
;
;     Symbol "INFO" not found in the JUPYTER package.
;
;       Line: 101, Column: 25, File-Position: 3797
;
;       Stream: #<SB-INT:FORM-TRACKING-STREAM for "file /var/tmp/portage/dev-python/maxima-jupyter-9999/work/maxima-jupyter-9999/src/kernel.lisp" {10071892F3}>
While evaluating the form starting at line 13, column 0

Maxima encountered a Lisp error:

 COMPILE-FILE-ERROR while compiling #<CL-SOURCE-FILE "maxima-jupyter" "kernel">

Automatically continuing.
To enable the Lisp debugger set *debugger-hook* to nil.
  of #P"/var/tmp/portage/dev-python/maxima-jupyter-9999/work/maxima-jupyter-9999/load-maxima-jupyter.lisp":;
                                                                                                           ; compilation unit aborted
                                                                                                           ;   caught 3 fatal ERROR conditions
                                                                                                           ;   caught 1 ERROR condition
Maxima encountered a Lisp error:

 The function MAXIMA-JUPYTER:KERNEL-START-EXEC is undefined.

Automatically continuing.
To enable the Lisp debugger set *debugger-hook* to nil.
(%o2) /var/tmp/portage/dev-python/maxima-jupyter-9999/temp/install-script.lisp

```

I am using latest repository version of maxima-jupyter with jupyter_core 4.4.0, sbcl 1.4.9 and maxima 5.42.1. This procedure was working in April but today is failing

--- a/src/kernel.lisp
+++ b/src/kernel.lisp
@@ -90,7 +90,7 @@
            (setq maxima::$_ maxima::$__)
            result)))))
 
-(defun read-and-eval (input in-maxima)
+(defun read-and-eval (kernel input in-maxima)
   (catch 'state-change
     (handling-errors
       (let ((code-to-eval (if in-maxima
@@ -98,7 +98,7 @@
                             (my-lread input))))
         (if code-to-eval
           (progn
-            (jupyter:info "[evaluator] Parsed expression to evaluate: ~W~%" code-to-eval)
+            (jupyter:inform :info kernel "Parsed expression to evaluate: ~W~%" code-to-eval)
             (when in-maxima
               (incf maxima::$linenum)
               (let ((label (maxima::makelabel maxima::$inchar)))
@@ -107,7 +107,7 @@
             (let ((result (if in-maxima
                             (my-eval code-to-eval)
                             (eval code-to-eval))))
-              (jupyter:info "[evaluator] Evaluated result: ~W~%" result)
+              (jupyter:inform :info kernel "Evaluated result: ~W~%" result)
               (when (and in-maxima (not (keyword-result-p result)))
                 (setq maxima::$% (caddr result)))
               result))
@@ -132,11 +132,11 @@
         (maxima::$stdin *query-io*)
         (maxima::$stderr *error-output*)
         (maxima::$stdout *standard-output*))
-    (jupyter:info "eval ~A~%" code)
+    (jupyter:inform :info k "eval ~A~%" code)
     (iter
       (with input = (make-string-input-stream code))
       (for in-maxima = (kernel-in-maxima k))
-      (for result = (read-and-eval input in-maxima))
+      (for result = (read-and-eval k input in-maxima))
       (until (eq result 'no-more-code))
       (for wrapped-result = (if in-maxima
                               (make-maxima-label result)
@@ -242,6 +242,6 @@
     (jupyter:handling-errors
       (multiple-value-bind (word start end) (symbol-string-at-position code cursor-pos)
         (when word
-          (jupyter::info "~A~%" word)
+          (jupyter:inform :info k "Inspect ~A~%" word)
           (make-instance 'inspect-result :symbol word))))
     (call-next-method)))

--- a/src/kernel.lisp
+++ b/src/kernel.lisp
@@ -21,7 +21,7 @@
   ((in-maxima :initform t
               :accessor kernel-in-maxima))
   (:default-initargs :name "maxima-jupyter"
-                     :package :maxima
+                     :package (find-package :maxima)
                      :version "0.7"
                      :banner "maxima-jupyter: a Maxima Jupyter kernel; (C) 2019 Robert Dodier (BSD)"
                      :language-name "maxima"

